### DISCUSSION TITLE
How is the process implemented after Elsa 3.1.3's designer process node is rejected for approval? Is the process moved back to the previous node or the original node?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5763] [General]
How is the process implemented after Elsa 3.1.3's designer process node is rejected for approval? Is the process moved back to the previous node or the original node?

**Comment by glime-ai** [comment]
When a process node in Elsa 3.1.3's designer is rejected for approval, the workflow can be configured to move back to the previous node or even to the original node, depending on the design of the workflow. To achieve this, you can use the "SignalReceived" activity to wait for a rejection signal and then use a combination of "Fork," "Join," and other control flow activities to route the workflow back to the desired node.

Here's a high-level approach to implement such logic:

1. Define a signal for rejection and for moving back to previous nodes.
2. Use "SignalReceived" activities to wait for these signals at appropriate points in your workflow.
3. Use "Fork" activities to create branches that handle the rejection.
4. Use "Join" activities to synchronize branches when necessary.
5. Connect the rejection signal to the desired previous node to allow the workflow to revert to an earlier state.

For example, if a node is rejected, you can have a "SignalReceived" activity waiting for a "Reject" signal. Upon receiving this signal, the workflow can transition back to the previous node or to the start of the process, depending on how you configure the flow.

Remember to persist the workflow state after each significant step so that you can revert to the correct state when moving back to a previous node.

For more detailed guidance and examples, you may refer to the documentation or community discussions related to Elsa Workflows.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
